src/test/java/teammates/test/driver/EmailAccount.java

```diff
@@ -26,73 +34,204 @@ private EmailAccount() {
+                || p.isMimeType("message/rfc822")
+                // as per the RFC2046 specification, other multipart subtypes are recognized as multipart/mixed
+                || p.isMimeType("multipart/*");
     }
 
-    private static Message[] getMessages(Folder inbox) throws MessagingException {
-        // Reading the Email Index in Read / Write Mode
-        inbox.open(Folder.READ_WRITE);
-        FlagTerm ft = new FlagTerm(new Flags(Flags.Flag.SEEN), false);
-        return inbox.search(ft);
+    private static String getTextFromMultiPartDigest(Part p) throws IOException, MessagingException {
+        Multipart mp = (Multipart) p.getContent();
+        StringBuilder textBuilder = new StringBuilder();
+        for (int i = 0; i < mp.getCount(); i++) {
+            Part bp = mp.getBodyPart(i);
+            if (bp.isMimeType("message/rfc822")) {
+                String text = getTextFromPart(bp);
+                if (text != null) {
+                    textBuilder.append(text);
+                }
+            }
+        }
+        String text = textBuilder.toString();
+
+        if (text.isEmpty()) {
+            return null;
+        }
+
+        return text;
     }
 
-    private static String getEmailBody(Message message) throws IOException, MessagingException {
-        String body = "";
+    /**
+     * Returns the text from multipart/alternative, the type of text returned follows the preference of the sending agent.
+     */
+    private static String getTextFromMultiPartAlternative(Part p) throws IOException, MessagingException {
+        Multipart mp = (Multipart) p.getContent();
 
-        if (message.getContent() instanceof String) {
-            body = message.getContent().toString();
-        } else if (message.getContent() instanceof Multipart) {
-            Multipart multipart = (Multipart) message.getContent();
-            BodyPart bodypart = multipart.getBodyPart(0);
-            body = bodypart.getContent().toString();
+        // if the multipart has no body parts, it is malformed
+        if (mp.getCount() == 0) {
+            return null;
         }
 
-        return body;
+        // search in reverse order as a multipart/alternative should have their most preferred format last
+        for (int i = mp.getCount() - 1; i >= 0; i--) {
+            Part bp = mp.getBodyPart(i);
+
+            if (bp.isMimeType("text/html")) {
+                return (String) bp.getContent();
+            } else if (bp.isMimeType("text/plain")) {
+                // since we are looking in reverse order, if we did not encounter a text/html first we can return the plain
+                // text because that is the best preferred format that we understand. If a text/html comes along later it
+                // means the agent sending the email did not set the html text as preferable or did not set their preferred
+                // order correctly, and in that case we do not handle that
```

Comment:
<!-- Capitalise the first letter of the first sentence and add a full stop here. -->